static/js/message_edit.js

```diff
@@ -185,6 +186,35 @@ export function get_deletability(message) {
     return false;
 }
 
+export function stream_and_topic_exist_in_edit_history(message, stream_id, topic) {
+    /*  Checks to see if a stream_id and a topic match any historical
+        stream_id and topic state in the message's edit history.
+     */
+    const narrow_dict = {stream_id, topic};
+    const message_dict = {stream_id: message.stream_id, topic: message.topic};
+
+    if (!message.edit_history) {
+        return false;
+    }
+
+    for (const edit_history_event of message.edit_history) {
+        if (
+            edit_history_event.prev_stream &&
+            edit_history_event.stream === message_dict.stream_id
+        ) {
+            message_dict.stream_id = edit_history_event.prev_stream;
+        }
+        if (edit_history_event.prev_topic && edit_history_event.topic === message_dict.topic) {
```

Comment:
Hmm, this comparison might need to be using the lower-case comparison functions too, right?

Reply:
Yes, correct. I also added that to where we compare the narrow topic and message topic in `adjusted_operators_if_moved`.
But that raises the question for me, when we compare the `message.display_recipient` and the narrow stream in `adjusted_operators_if_moved`, do we also want to do so in lowercase?

Reply:
Yes, that sounds right; posted a comment in place there.